### ISSUE TITLE
Hex editor should be accessible

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode-hexeditor#4]
Hey Logan, just kreating this so we keep it in mind.
The hex editor should be accessible, that means that all fields should be navigalbe by keyboard and that when you navigate the screen reader always reads the correct content.

For more details we can also sync. Feel free to ping me

**Comment by lramos15** [comment]
This is a large debt that I'm currently having issues working on. Whenever I spin up NVDA it attempts to read the entire DOM which then crashes VS Code.

**Comment by isidorn** [comment]
@lramos15 even without NVDA you can work on keyboard focus and navigation.

As for NVDA problems make sure you have latest and if you have steps, just file an issue to their repository as they are very responsive and might help you out https://github.com/nvaccess/nvda

Thanks for looking into this

**Comment by lramos15** [comment]
I should say I don't think this is an NVDA specific issue as it happens with windows narrator as well, it's something with the way the DOM is laid out on the editor. I will spend debt week looking into making sure the editor is easy to navigate with the keyboard.

**Comment by isidorn** [comment]
@lramos15 yeah starting with the keyboard makes sense.
Sorry I misunderstood you originaly, I thought you see the issue everytime you open any editor. But you only see it when you open your Hex editor. Is the Hex editor virtualised? Do we plan to do that? Asking since this might help.

**Comment by lramos15** [comment]
It is virtualized already. I don't know why it does that. Maybe we can sync up and I can show you after the keyboard navigation work is done.

**Comment by isidorn** [comment]
Sure, but keep in mind that next three weeks I am on vacation.

**Comment by zersiax** [comment]
Curious what happened to this in the end? I could use an accessible hex editor :)

**Comment by isidorn** [comment]
@lramos15 might have updates here

**Comment by lramos15** [comment]
Hi @zersiax, 
Ideally this is still something we want to do in the future. When I was building the hex editor I had problems with NDA crashing due to my virtual DOM construction and therefore wasn't able to bake accessibility in while building the product. I started this extension as an intern at Microsoft and now am back full time which is why you see a gap in the contribution log. This task isn't forgotten and is one of the few issues on the priority list that you see pinned in the issue list. Let me know if you have any other questions or any small changes I could to make to the hex editor to make it more usable for you as a fully accessible one might take some time, but I may be able to improve the experience in an iterative fashion.

**Comment by zersiax** [comment]
I just had a quick play with it, here are my findings:

- With NVDA, in focus mode, you can tab and arrow through the values. Arrows alter the character you're on, tab seems to flip between representations. I tested this with a text file with just the word "hello" in it, and that seemed to work ok, although I will admit I am very much new to hex editing. I have no idea how practical this way of working would be in a larger file.
- - I can't seem to figure out how to edit a value, I can look at them but not change them for as far as I can see.
- There's a listing of all the hex values before the actual file contents "00 01 02 03 04 05 etc), I'm not entirely sure what that is supposed to be indicating :)
So yeah, some things work, others I just have no idea about :)

**Comment by Tyriar** [comment]
@lramos15 FYI I've gone through a lot of the challenges you would probably have and spoken with a host of people about this for the terminal which is also virtualized in a similar way. You might get some insights from reading through some of the issues in https://github.com/microsoft/vscode/issues?q=is%3Aissue+label%3Aintegrated-terminal+label%3Aaccessibility+is%3Aclosed but I'm sure I can make some time to talk about this.

**Comment by sukiletxe** [comment]
> There's a listing of all the hex values before the actual file contents "00 01 02 03 04 05 etc), I'm not entirely sure what that is supposed to be indicating :)

Seems like columns. Default is 16 bytes per row, that's why it goes from 00 to F.
